roles/prometheus/templates/prometheus.service.j2

```diff
@@ -13,12 +13,16 @@ User=prometheus
 Group=prometheus
 ExecReload=/bin/kill -HUP $MAINPID
 ExecStart={{ _prometheus_binary_install_dir }}/prometheus \
+{% if not prometheus_agent_mode %}
```

Comment:
we can't use this arguments with agent mode, getting the following error:
```Aug 17 20:13:35 master2-int prometheus[50202]: The following flag(s) can not be used in agent mode: ["--storage.tsdb.path" "--storage.tsdb.retention.time" "--storage.tsdb.retention.size"]```

Reply:
The flags are different for agent mode. `--storage.agent.path`, `--storage.agent.retention.min-time`, `--storage.agent.retention.max-time`.

Reply:
cool, now `--storage.agent.path={{ prometheus_db_dir }}` is in place. As for `--storage.agent.retention.min-time` and `--storage.agent.retention.max-time` I guess people could use `prometheus_config_flags_extra`

Reply:
We could do a var in `vars/main.yaml` that is something like this:

```jinja
_prometheus_storage_mode: "{{ "agent" if prometheus_agent_mode else "tsdb" }}"
```

Then the flag template would be:
```jinja
  --storage.{{ _prometheus_storage_mode }}.path={{ prometheus_db_dir }}
```

Reply:
@SuperQ I'm trying to understand what the benefit we get from that? I mean we still need to
1. Include `--enable-feature=agent \` to the template
2. Exclude (remove) the flags `--storage.tsdb.retention.time` and `--storage.tsdb.retention.size`
So that being said, we still need to define `if` statement with `prometheus_agent_mode` in the template regardless of using either `--storage.{{ _prometheus_storage_mode }}.path` or `--storage.agent.path`/`--storage.tsdb.path`

Reply:
`--storage.{{ _prometheus_storage_mode }}.retention.time`

Reply:
unfortunately not going to work
```
Aug 20 05:58:07 ip-10-16-147-81 prometheus[8786]: Error parsing command line arguments: unknown long flag '--storage.agent.retention.time'
Aug 20 05:58:07 ip-10-16-147-81 prometheus[8786]: prometheus: error: unknown long flag '--storage.agent.retention.time'
```

Reply:
Oh, right, I wasn't looking at the agent flags closely enough.